src/FakeItEasy.Analyzer/RepeatedAssertionAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+namespace FakeItEasy.Analyzer
+{
+    using System;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+#if CSHARP
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+#endif
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.Text;
+#if VISUAL_BASIC
+    using Microsoft.CodeAnalysis.VisualBasic;
+    using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+#endif
+
+#if CSHARP
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+#elif VISUAL_BASIC
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+#endif
+    public class RepeatedAssertionAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(DiagnosticDefinitions.RepeatedAssertion);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.RegisterSyntaxNodeAction(AnalyzeCall, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeCall(SyntaxNodeAnalysisContext context)
+        {
+            if (!(context.Node is InvocationExpressionSyntax call))
+            {
+                return;
+            }
+
+            if (call.ArgumentList.Arguments.Count != 1)
+            {
+                return;
+            }
+
+            var methodSymbol = SymbolHelpers.GetCalledMethodSymbol(call, context);
+            if (methodSymbol?.GetFullName() != "FakeItEasy.Configuration.IAssertConfiguration.MustHaveHappened")
+            {
+                return;
+            }
+
+            // call is an InvocationExpression, with two children - a SimpleMemberAccessExpression
+            // (like A.CallTo(() => fake.Method()).MustHaveHappened), and an argument list.
+            // The SimpleMemberAccessExpression's first child is the IAssertConfiguration (for example,
+            // A.CallTo(() => fake.Method())), and the second is the IdentifierName MustHaveHappened.
+            var memberAccessNode = call.ChildNodes().FirstOrDefault();

```

Comment:
[`call.Expression`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntax.invocationexpressionsyntax.expression?view=roslyn-dotnet) would achieve the same result, and doesn't rely on the order of child nodes.

Reply:
Ha. I'd figured this sort of thing out when working on the code fix and hadn't brung it back. Thanks!